api/types/resource.go

```diff
@@ -452,6 +495,56 @@ func (m *Metadata) SetOrigin(origin string) {
 	m.Labels[OriginLabel] = origin
 }
 
+// IsEqual determines if two metadata resources are equivalent to one another.
+func (m *Metadata) IsEqual(i interface{}) bool {
+	ptr, ok := i.(*Metadata)
+	if ok {
+		if m == nil && ptr == nil {
+			return true
+		}
+		if m != nil && ptr != nil {
+			return m.isEqual(*ptr)
+		}
+		return false
+	}
```

Comment:
This common logic repeats in every equality check function, I wonder if we can pull it out into a reusable helper method.